### DISCUSSION TITLE
Is there any way to parse to a specific tick as fast as ParseToEnd()?

### DISCUSSION THREAD

**Discussion by srjnm** [markus-wa/demoinfocs-golang#345] [Q&A] [answered]
Right now I'm parsing to a specific tick using a for loop and `ParseNextFrame()`
```
f, err := os.Open(fileName)
if err != nil {
    log.Println(err)
}
defer f.Close()

r := dem.NewParser(f)

for r.GameState().IngameTick() < endTick) {
    _, err := r.ParseNextFrame()
    if err != nil {
	log.Println(err.Error())
    }
}
```
This code is really slow unlike using `ParseToEnd()`.
Is there any alternative to this like,
- Start parsing from specific tick
- Move to specific tick without parsing anything and then `ParseToEnd()`

**Comment by markus-wa** [comment, marked as answer]
Hi @srjnm 

Great question!

Basically I would never recommend using `ParseNextFrame()`.

Instead you should

- A: use `RegisterEventHandler()` and `UnregisterEventHandler()` 
- or B: handle events conditionally in your code based on  if you are in the right tick

also, `r.GameState().IngameTick() < endTick` may not work, as the last tick from the header is often not correct

hope that makes sense

**Reply by markus-wa** [reply]
and just to clarify, `ParseNextFrame()` can never be as fast as `ParseToEnd()` as it gives different guarantees and therefore can't parse ahead of the event that is being handled
